sentence.go

```diff
@@ -74,69 +79,137 @@ func (s BaseSentence) TalkerID() string {
 // String formats the sentence into a string
 func (s BaseSentence) String() string { return s.Raw }
 
-// parseSentence parses a raw message into it's fields
-func parseSentence(raw string) (BaseSentence, error) {
+// SentenceParser is configurable parser instance to parse raw input into NMEA0183 Sentence
+//
+// SentenceParser fields/methods are not co-routine safe!
+type SentenceParser struct {
+	// CustomParsers allows registering additional parsers
+	CustomParsers map[string]ParserFunc
+
+	// ParsePrefix takes in the sentence first field (NMEA0183 address) and splits it into a talker id and sentence type
+	ParsePrefix func(prefix string) (talkerID string, sentence string, err error)
+
+	// CheckCRC allows custom handling of checksum checking based on parsed sentence
+	CheckCRC func(sentence BaseSentence, rawFields string) error
+
+	// OnTagBlock is callback to handle all parsed tag blocks even for lines containing only a tag block and
+	// allows to track multiline tag group separate lines. Logic how to combine/assemble multiline tag group
+	// should be implemented outside SentenceParser
+	//
+	// Example of group of 3:
+	// \g:1-3-1234,s:r3669961,c:1120959341*hh\
+	// \g:2-3-1234*hh\!ABVDM,1,1,1,B,.....,0*hh
+	// \g:3-3-1234*hh\$ABVSI,r3669961,1,013536.96326433,1386,-98,,*hh
+	OnTagBlock func(tagBlock TagBlock)
+}
+
+// ParseBaseSentence parses BaseSentence from input
+// This method is not co-routine safe!
+func (p *SentenceParser) ParseBaseSentence(raw string) (BaseSentence, error) {
```

Comment:
I don't think `ParseBaseSentence` needs to be exported either. No one has ever asked for `parseSentence` to be exported.

Reply:
I'll remove that. I think this is leftover from my initial idea, before `SentenceParser` struct existed, to expose some of the parsing functions so you could compose your own parser from these "fragments"